src/Roslyn.Diagnostics.Analyzers/UnitTests/CSharpDoNotCapturePrimaryConstructorParametersTests.cs

```diff
@@ -0,0 +1,213 @@
+                    }
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp12,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ErrorOnCapture_UseInSubsequentConstructor()
+        {
+            var source = """
+                class C(int i)
+                {
+                    C(bool b) : this(1)
+                    {
+                        _ = i;
+                    }
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp12,
+                ExpectedDiagnostics = {
+                    // /0/Test0.cs(5,13): error RS0103: Primary constructor parameter 'i' should not be implicitly captured
+                    VerifyCS.Diagnostic().WithSpan(5, 13, 5, 14).WithArguments("i"),
+                    // /0/Test0.cs(5,13): error CS9105: Cannot use primary constructor parameter 'int i' in this context.
+                    DiagnosticResult.CompilerError("CS9105").WithSpan(5, 13, 5, 14).WithArguments("int i"),
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NoError_PassToBase()
+        {
+            var source = """
+                class Base(int i);
+                class Derived(int i) : Base(i);
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp12,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NoError_FieldInitializer()
+        {
+            var source = """
+                class C(int i)
+                {
+                    public int I = i;
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp12,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NoError_PropertyInitializer()
+        {
+            var source = """
+                class C(int i)
+                {
+                    public int I { get; set; } = i;
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                LanguageVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp12,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NoError_CapturedInLambda()
+        {
+            var source = """
+                using System;
+                public class Base(Action action);
+                public class Derived(int i) : Base(() => Console.WriteLine(i));
```

Comment:
Checking my understanding: when `i` is used inside the lambda here, is the `context.ContainingSymbol` still `Derived..ctor(int)`?

Reply:
Correct.